web/src/compose_reply.js

```diff
@@ -114,13 +114,21 @@ export function reply_with_mention(opts) {
 }
 
 export function quote_and_reply(opts) {
-    const $textarea = $("#compose-textarea");
+    let $textarea = $("#compose-textarea");
     const message_id = opts.message_id || message_lists.current.selected_id();
     const message = message_lists.current.get(message_id);
     const quoting_placeholder = $t({defaultMessage: "[Quoting…]"});
 
-    if (!compose_state.has_message_content()) {
+    // If we are editing another message (which is selected), we want to add the quote
+    // in the edit textarea and not the compose box.
+    const editing = message_lists.current.selected_row().find(".message_edit_content").length;
+    if (editing) {
+        $textarea = message_lists.current.selected_row().find(".message_edit_content");
+    }
+
```

Comment:
I think there's a question of whether we want to do this or instead we want to ignore the selected message and just react to the last message edit area or compose area that was interacted with (which we can keep track of with an on-`focus` event handler registered for the message edit and compose textareas).

Reply:
Have added a prep commit to keep track of the last focused compose style textarea in `compose_state`, and then if that textarea is still connected to the DOM (which is not the case when the last focused message being edited is Cancelled or Saved before quoting) the message is quoted into that textarea, otherwise defaulting to the main compose textarea.